### PR TITLE
[Data Virtualization]: Create datastores with groupId via aqueduct

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
[AB#7382](https://dev.azure.com/fluidframework/internal/_workitems/edit/7382)

Enable `PureDataObjectFactory` to create `PureDataObject`s with loadingGroupId

### CODE REVIEW

File: packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts [hunk 1]
```diff
@@ -214,17 +214,20 @@ export class PureDataObjectFactory<
 	 * @param context - The context being used to create the runtime
 	 * (the created object will have its own new context created as well)
 	 * @param initialState - The initial state to provide to the created data store.
+	 * @param loadingGroupId - experimental feature, that tries to delay load a data store. The service needs to support this feature, does not work for most services
```
Comment (vladsud) [inline packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts:217]: Should we use @experimantal tag here? Can it apply to an argument?

Comment (vladsud) [reply packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts:217]: Also please make sure everywhere where this argument shows up, we make a reference to FluidFramework/packages/runtime/container-runtime/README.md

File: packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts [hunk 2]
```diff
@@ -214,17 +214,20 @@ export class PureDataObjectFactory<
...
 	 * @returns an object created by this factory. Data store and objects created are not attached to container.
 	 * They get attached only when a handle to one of them is attached to already attached objects.
 	 */
 	public async createChildInstance(
 		parentContext: IFluidDataStoreContext,
 		initialState?: I["InitialState"],
+		loadingGroupId?: string,
```
Comment (vladsud) [inline packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts:224]: One of earlier pieces of feedback we received - is it required to expose loadingGroupID, or can we use a boolean (load or not to load, even if under the cover we generate loadingGroup=uuid()).
I think ability to pass ID is required, as I think usage pattern will be - passing same ID for many data stores, but worth confirming.

**Review by vladsud — APPROVED** [review]
Let's do some due diligence to ensure this is the required form of API that Loop can leverage before merging.

File: packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts (hunk 2 above)
Comment (tyler-cai-microsoft) [reply packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts:224]: Offline, we confirmed that customers want to control loading group id. We cannot surface just use a boolean for them.

File: packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts (hunk 1 above)
Comment (tyler-cai-microsoft) [reply packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts]: Added the package reference. When I wrote `@experimental` the comment was lost in the tsdoc.
This is what it looks like now:
image

This is what it looks like if I try to add `@experimental`
image

**Pull request merged**